src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/ProgrammableInvoker.java

```diff
@@ -107,12 +108,12 @@
 
     private final long bufferCopySize;
 
-    public ProgrammableInvoker(ABIDescriptor abi, MemoryAddress addr, CallingSequence callingSequence) {
+    public ProgrammableInvoker(ABIDescriptor abi, Addressable addr, CallingSequence callingSequence) {
         this.abi = abi;
         this.layout = BufferLayout.of(abi);
         this.stubAddress = adapterStubs.computeIfAbsent(abi, key -> generateAdapter(key, layout));
 
-        this.addr = addr;
+        this.addr = addr.address();
```

Comment:
Don't think this will keep the reference to the LibraryLookup alive. I think you need to save the Addressable, and then convert to MA on demand, or use 2 fields (or maybe you have a better idea).

Reply:
Gosh - you are right; what I did in the addressable patch was to store the addressable as-is in the programmable invoker field; I should just do that.